src/map/enmity_container.cpp

```diff
@@ -102,6 +103,27 @@ float CEnmityContainer::CalculateEnmityBonus(CBattleEntity* PEntity){
     {
         enmityBonus = ((CCharEntity*)PEntity)->PMeritPoints->GetMeritValue(MERIT_ENMITY_INCREASE, (CCharEntity*)PEntity) -
             ((CCharEntity*)PEntity)->PMeritPoints->GetMeritValue(MERIT_ENMITY_DECREASE, (CCharEntity*)PEntity);
+
+		// Handle stealth shot.
+		ACTIONTYPE action = PEntity->PBattleAI->GetCurrentAction();
+		if (action == ACTION_RANGED_FINISH ||
+			action == ACTION_WEAPONSKILL_FINISH) {
+
+			CStatusEffect* stealthShot = PEntity->StatusEffectContainer->GetStatusEffect(EFFECT_STEALTH_SHOT);
+			if (stealthShot != nullptr) {
+
+				// Ranged attacks, and all ranged weapon skills besides Trueflight should get a bonus applied.
+				if (action == ACTION_RANGED_FINISH ||
+					((PEntity->PBattleAI->GetCurrentWeaponSkill()->getType() == 25 ||
+					PEntity->PBattleAI->GetCurrentWeaponSkill()->getType() == 26) &&
+					PEntity->PBattleAI->GetCurrentWeaponSkill()->getID() != 217)) {

```

Comment:
this is a pretty good indication that this should be handled in the script level instead of here (which may not be fully supported yet - in which case it's better to leave this unimplemented until it is)